src/api/worker/rest/DefaultEntityRestCache.ts

```diff
@@ -111,6 +114,11 @@ export interface ExposedCacheStorage {
 	getLastUpdateTime(): Promise<LastUpdateTime>
 
 	clearExcludedData(): Promise<void>
+
+	/**
+	 * remove an entity from the cache by typeRef and Id
+	 * */
+	deleteIfExists<T extends SomeEntity>(typeRef: TypeRef<T>, listId: Id | null, id: Id): Promise<void>
```

Comment:
Ideally we shouldn't allow deleting `ListElement`s willy-nilly but maybe if we mention it in the jsdoc it is enough?

Reply:
I think it would be good to mention that, yes

we can change the signature to always take null as listId and make T extends ElementEntity as well, the implementation/internal interface should be allowed to take wider types while the exposed interface is more restricted?